Pod/Classes/LSClockState.m

```diff
@@ -1,148 +1,93 @@
 #import "LSClockState.h"
 
+#pragma mark - Constants
+
 static const int kMaxOffsetAge = 7;
 static const UInt64 kStoredSamplesTTLMicros = 60 * 60 * 1e6;
+static NSString *kDelayKey = @"delay";
+static NSString *kOffsetKey = @"offset";
+static NSString *kUserDefaultsKey = @"com.lightstep.clock_state";
+static NSString *kTimestampMicrosKey = @"timestamp_micros";
+static NSString *kSamplesKey = @"samples";
+
+#pragma mark - LSClockSample
 
 @interface LSSyncSample : NSObject<NSCoding>
 
-- (id) initWithDelayMicros:(SInt64)delayMicros offsetMicros:(SInt64)offsetMicros;
+@property(nonatomic) SInt64 delayMicros;
+@property(nonatomic) SInt64 offsetMicros;
 
-@property (nonatomic) SInt64 delayMicros;
-@property (nonatomic) SInt64 offsetMicros;
+- (id)initWithDelayMicros:(SInt64)delayMicros offsetMicros:(SInt64)offsetMicros;
 
 @end
 
 @implementation LSSyncSample
 
-- (id) initWithDelayMicros:(SInt64)delayMicros offsetMicros:(SInt64)offsetMicros
-{
+- (id)initWithDelayMicros:(SInt64)delayMicros offsetMicros:(SInt64)offsetMicros {
     if (self = [super init]) {
-        self.delayMicros = delayMicros;
-        self.offsetMicros = offsetMicros;
+        _delayMicros = delayMicros;
+        _offsetMicros = offsetMicros;
     }
     return self;
 }
 
-static NSString* kDelayKey = @"delay";
-static NSString* kOffsetKey = @"offset";
-
-- (id) initWithCoder:(NSCoder* )aDecoder {
-    return [self initWithDelayMicros:[aDecoder decodeInt64ForKey:kDelayKey] offsetMicros:[aDecoder decodeInt64ForKey:kOffsetKey]];
+- (id)initWithCoder:(NSCoder *)aDecoder {
+    return [self initWithDelayMicros:[aDecoder decodeInt64ForKey:kDelayKey]
+                        offsetMicros:[aDecoder decodeInt64ForKey:kOffsetKey]];
 }
 
-- (void) encodeWithCoder:(NSCoder* )aCoder
-{
+- (void)encodeWithCoder:(NSCoder *)aCoder {
     [aCoder encodeInt64:self.delayMicros forKey:kDelayKey];
     [aCoder encodeInt64:self.offsetMicros forKey:kOffsetKey];
 }
 
 @end
 
-@implementation LSClockState {
-    NSMutableArray* m_samples;  // elements are LSSyncSamples
-    SInt64 m_currentOffsetMicros;
-    int m_currentOffsetAge;
-}
+#pragma mark - LSClockState
+
+@interface LSClockState ()
+@property(nonatomic, strong) NSMutableArray<LSSyncSample *> *samples;
+@property(nonatomic) NSInteger currentOffsetAge;
+@property(nonatomic) SInt64 currentOffsetMicros;
```

Comment:
Over the wire we will use epoch microseconds to communicate with LightStep (our lingua franca)... OpenTracing uses `NSDate *` as you know. Between OpenTracing and the LightStep wire format I think it'll be easier to just use microseconds everywhere, but it's not something I feel strongly about.